BaragonCore/src/main/java/com/hubspot/baragon/models/BaragonGroup.java

```diff
@@ -1,34 +1,46 @@
 package com.hubspot.baragon.models;
 
 import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
 import java.util.Set;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonProperty;
-import com.google.common.base.Objects;
+import com.google.common.base.MoreObjects;
 import com.google.common.base.Optional;
 
 @JsonIgnoreProperties( ignoreUnknown = true )
 public class BaragonGroup {
   private final String name;
   @Deprecated
   private Optional<String> domain;
-  private Set<TrafficSource> sources;
+  private Set<TrafficSource> trafficSources;
+  @Deprecated
+  private Set<String> sources;
   private Optional<String> defaultDomain;
   private Set<String> domains;
 
   @JsonCreator
   public BaragonGroup(@JsonProperty("name") String name,
                       @JsonProperty("domain") Optional<String> domain,
-                      @JsonProperty("sources") Set<TrafficSource> sources,
+                      @JsonProperty("trafficSources") Set<TrafficSource> trafficSources,
+                      @JsonProperty("sources") Set<String> sources,
                       @JsonProperty("defaultDomain") Optional<String> defaultDomain,
                       @JsonProperty("domains") Set<String> domains) {
     this.name = name;
     this.domain = domain;
-    this.sources = Objects.firstNonNull(sources, Collections.<TrafficSource>emptySet());
     this.defaultDomain = defaultDomain;
-    this.domains = Objects.firstNonNull(domains, Collections.<String>emptySet());
+    this.domains = MoreObjects.firstNonNull(domains, Collections.<String>emptySet());
+
+    if (sources == null) {
```

Comment:
I think we can be a bit more clear about what is going on here, it's a bit hard to reason about this way. Maybe something like this?
```java
    this.sources = MoreObjects.firstNonNull(sources, Collections.<String>emptySet());
    if (trafficSources == null) {
      this.trafficSources = sources.forEach((source) -> new TrafficSource(source, TrafficSourceType.CLASSIC));
    } else {
      this.trafficSources = trafficSources;
    }
```
might have to make sure the most recent stuff is merged in to use the java8 bits, but it would be cleaner/easier to read without having to have the static methods